### PR TITLE
Added History As an action

### PR DESCRIPTION
**Pull request by Priyansh910**

### CODE REVIEW

File: src/lib/components/Grid/index.js [hunk 1]
```diff
@@ -214,15 +216,15 @@ const GridBase = memo(({
...
+    const userDefinedPermissions = { add: effectivePermissions.add || true, edit: effectivePermissions.edit, delete: effectivePermissions.delete };
     const { canAdd, canEdit, canDelete } = getPermissions(userData, model, userDefinedPermissions);
     const filterFieldDataTypes = {
         Number: 'number',
         String: 'string',
         Boolean: 'boolean'
     };
 
-    const { addUrlParamKey, searchParamKey, hideBreadcrumb = false } = model;
+    const { addUrlParamKey, searchParamKey, hideBreadcrumb = false, tableName, canHistory = true, gridTitle } = model;
```
Comment (manojpandeyDev18) [inline src/lib/components/Grid/index.js:227]: canHistory --> should be showHistory

File: src/lib/components/Grid/index.js [hunk 2]
```diff
@@ -214,15 +216,15 @@ const GridBase = memo(({
     const tablePreferenceEnums = stateData?.gridSettings?.permissions?.tablePreferenceEnums;
     const emptyIsAnyOfOperatorFilters = ["isEmpty", "isNotEmpty", "isAnyOf"];
     const userData = stateData.getUserData;
-    const userDefinedPermissions = { add: effectivePermissions.add, edit: effectivePermissions.edit, delete: effectivePermissions.delete };
+    const userDefinedPermissions = { add: effectivePermissions.add || true, edit: effectivePermissions.edit, delete: effectivePermissions.delete };
```
Comment (manojpandeyDev18) [inline src/lib/components/Grid/index.js:219]: why by default true here ?

File: src/lib/components/Grid/index.js (hunk 1 above)
Comment (Priyansh910) [reply src/lib/components/Grid/index.js]: Changed to showHistory

File: src/lib/components/Grid/index.js (hunk 2 above)
Comment (Priyansh910) [reply src/lib/components/Grid/index.js]: Removed default true from add

**Review by manojpandeyDev18 — APPROVED** [review]

**Pull request merged**